src/cascadia/TerminalApp/TerminalPaneContent.idl

```diff
@@ -2,15 +2,15 @@
 // Licensed under the MIT license.
 
 import "IPaneContent.idl";
+import "TerminalSettingsCache.idl";
 
 namespace TerminalApp
 {
     [default_interface] runtimeclass TerminalPaneContent : IPaneContent, ISnappable
     {
         Microsoft.Terminal.Control.TermControl GetTermControl();
 
-        void UpdateSettings(const Microsoft.Terminal.Settings.Model.TerminalSettingsCreateResult settings,
-                            const Microsoft.Terminal.Settings.Model.Profile profile);
+        void UpdateTerminalSettings(TerminalSettingsCache cache);

```

Comment:
From an "offline" (lol) discussion:
It may make sense here to construct `TerminalPaneContent` with a shared reference to the `TerminalSettingsCache` and then let `UpdateSettings()` create the cached objects lazily. That way we avoid the extra API function and `try_as` tests.

We can do this in a follow-up PR.